### PR TITLE
[JENKINS-58362] Need to have only one copy of SystemCommandLanguage among detached plugins

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-58362](https://issues.jenkins-ci.org/browse/JENKINS-58362)

After https://github.com/jenkinsci/command-launcher-plugin/pull/4 and https://github.com/jenkinsci/jenkins/pull/4000/commits/b93ab72e6796e2bd611b14e86b262f897a01605f in #4000 we could have two copies of `SystemCommandLanguage` in the system, leading to weird situations. Fixes

```
mvn -f ../command-launcher-plugin surefire:test -Djenkins.version=2.184-SNAPSHOT -Dtest=CommandLauncher2Test\#requireApproval
```

since otherwise the test loads an old detached version of `command-launcher` (due to another bug: #4100) and we get

```
expected:<org.jenkinsci.plugins.scriptsecurity.scripts.languages.SystemCommandLanguage@…> but was:<org.jenkinsci.plugins.command_launcher.SystemCommandLanguage@…>
```

Yes this would have been caught some time ago by running `plugin-compat-tester` regularly.

Proposed changelog entry:
* Due to a mixture of old and new “detached” plugins, under certain circumstances it was possible to have two versions of an extension point used by the command launcher option for agents.

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]

File: war/pom.xml [hunk 1]
```diff
@@ -405,7 +405,7 @@ THE SOFTWARE.
                 <artifactItem>
                   <groupId>org.jenkins-ci.plugins</groupId>
                   <artifactId>command-launcher</artifactId>
-                  <version>1.0</version>
+                  <version>1.2</version>
```
Comment (jglick) [inline war/pom.xml]: [Diff](https://github.com/jenkinsci/command-launcher-plugin/compare/command-launcher-1.0...command-launcher-1.2)

**Review by raul-arabaolaza — APPROVED** [review]
This fixes some broken internal PCT runs

**Comment by jglick** [comment]
@olivergondza I am proposing this for backport to 2.176.x, maybe to .2 if it is not too late.

File: test/src/test/java/jenkins/install/LoadDetachedPluginsTest.java [hunk 2]
```diff
@@ -87,62 +86,6 @@ public void upgradeFromJenkins2() {
         });
     }
 
-    @Issue("JENKINS-48604")
-    @Test
-    @LocalData
-    public void upgradeFromJenkins2WithNewerDependency() {
```
Comment (jglick) [inline test/src/test/java/jenkins/install/LoadDetachedPluginsTest.java]: @dwnusbaum FYI I think these were your tests. Not sure how to rework them to be less brittle.

Comment (dwnusbaum) [reply test/src/test/java/jenkins/install/LoadDetachedPluginsTest.java]: Yeah, I don't have any good idea myself. The tests either need to use actual detached plugins or dynamically set up some simple, self-contained plugins as if they were detached plugins and use those in tests. The first option is brittle, and the second seems complicated.

Comment (jglick) [reply test/src/test/java/jenkins/install/LoadDetachedPluginsTest.java]: Right, the second option sort of presumes a test harness that would let you work with a specially-built WAR file. Beyond the power of `JenkinsRule`.

**Review by olivergondza — APPROVED** [review]

**Comment by olivergondza** [comment]
@jglick, lets get this integrated into master, first. I do not see a reason this would not make it to .3, but you are right is is too late for .2.

**Comment by oleg-nenashev** [comment]
@jglick  please write a changelog entry for the fix, see the PR template.
I will merge this change tomorrow if no negative feedback

**Pull request merged**